uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-keycloak/src/main/java/org/uberfire/ext/security/management/keycloak/BaseKeyCloakManager.java

```diff
@@ -165,7 +174,13 @@ protected Group createGroup(String name) {
 
     protected Set[] getUserGroupsAndRoles(final RoleMappingResource roleMappingResource) {
         if (roleMappingResource != null) {
-            List<RoleRepresentation> roles = roleMappingResource.realmLevel().listEffective();
+            List<RoleRepresentation> roles;
+            if (useResourceRoleMapping) {
```

Comment:
I saw this IF Pattern 3 times. Can't we do something to extract that first part?

Something like: 

```
public ACommonInterface getSomething(){
  if (useResourceRoleMapping) {
    return roleMappingResource.clientLevel(getClientIdByName(getRealmResource()))
   } else {
     return roleMappingResource.realmLevel()
   }
}
```

Reply:
@adrielparedes I have refractored the code. Thanks